### PR TITLE
Test/testing module

### PR DESCRIPTION
**Pull request by Se3do**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Other... Please describe: unit tests

## What is the current behavior?
The `@nestjs/testing` package currently has no unit tests for its public API classes (`Test`, `TestingModuleBuilder`, `TestingModule`, `TestingLogger`, `TestingInjector`, `TestingInstanceLoader`).

Issue Number: N/A

## What is the new behavior?
Adds 44 unit tests across 5 spec files achieving coverage of all public methods in the testing package:

- **`test.spec.ts`** (2 tests) — `Test.createTestingModule` with and without options
- **`testing-logger.service.spec.ts`** (5 tests) — all `TestingLogger` methods, verifies silent log/warn/debug/verbose and error delegation to `ConsoleLogger`
- **`testing-module-builder.spec.ts`** (17 tests) — constructor, all 6 override methods (shape + chaining), `useMocker`, `setLogger`, `compile` with overrides, mocker fallback, snapshot/preview options
- **`testing-module.spec.ts`** (6 tests) — constructor, `createNestMicroservice`, `createNestApplication` (adapter proxy + logger options), `get`, `select`, `init`/`close` lifecycle
- **`testing-injector.spec.ts`** (10 tests) — `TestingInjector` (setMocker, setContainer, resolveComponentWrapper with 4 paths, resolveComponentHost with 3 paths), `TestingInstanceLoader` (with/without mocker, super delegation)

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information
All tests use mocha + chai + sinon, follow existing NestJS test conventions, and include proper `afterEach` cleanup for spies/stubs and global state (`UuidFactory.mode`).

### CODE REVIEW

File: packages/testing/test/testing-module.spec.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    it('should extend NestApplicationContext', () => {
+      const container = new NestContainer();
+      const rootModule = new Module(class _RootTest {}, container);
+
+      const module = new TM(
+        container,
+        NoopGraphInspector,
+        rootModule,
+        new ApplicationConfig(),
+      );
```
[the remark this answers is not in the document]
Comment (Se3do) [reply packages/testing/test/testing-module.spec.ts:50]: `NoopGraphInspector` is a Proxy constant (not a class), typed as `GraphInspector`. Passing it directly is correct - see packages/core/inspector/noop-graph-inspector.ts.

File: packages/testing/test/testing-module.spec.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      const module = await Test.createTestingModule({
+        imports: [TestAppModule],
+      }).compile();
+
+      const overrideSpy = sinon.spy(Logger, 'overrideLogger');
+
+      const adapter = createMockAdapter();
+      module.createNestApplication(adapter as any, { logger: false });
+
+      expect(overrideSpy.calledOnce).to.be.true;
```
[the remark this answers is not in the document]
Comment (Se3do) [reply packages/testing/test/testing-module.spec.ts:112]: `Logger.overrideLogger()` is already restored by the `afterEach(() => sinon.restore())` at line 36, and every subsequent `compile()` call re-applies the logger via `TestingModuleBuilder.applyLogger()`. No leak.

File: packages/testing/test/testing-logger.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe('log', () => {
+    it('should not write to stdout', () => {
+      const spy = sinon.spy(process.stdout, 'write');
+      logger.log('test message');
+      expect(spy.called).to.be.false;
+    });
```
[the remark this answers is not in the document]
Comment (Se3do) [reply packages/testing/test/testing-logger.service.spec.ts:22]: These methods are explicit no-ops - they have empty bodies:
```
log(message: string) {}
warn(message: string) {}
debug(message: string) {}
verbose(message: string) {}
```
They cannot write to any stream by definition. The stdout spy correctly verifies the no-output contract. Adding stderr coverage would test an impossibility since the methods are empty.
